contrib/pyln-client/tests/test_units.py

```diff
@@ -269,3 +269,17 @@ def test_nonegative():
         Millisatoshi("1sat") // -1
     with pytest.raises(ValueError, match='Millisatoshi must be >= 0'):
         Millisatoshi("1btc") // -1
+
+
+def test_mul():
+    amount = Millisatoshi(21) * 2
+    assert amount == Millisatoshi(42)
+    amount = Millisatoshi(21) * Millisatoshi(2)
+    assert amount == Millisatoshi(42)
+
+
+def test_div():
+    amount = Millisatoshi(42) / 2
+    assert amount == Millisatoshi(21)
+    amount = Millisatoshi(42) / Millisatoshi(2)
+    assert amount == Millisatoshi(21)
```

Comment:
The division between Millisatoshis makes sense, but can we check the type here? For example in this context:
```
our_liquidity = Millisatoshi(5)
channel_size = Millisatoshi(10)
ratio = our_liquidity/channel_size
```
The return type should be a `float`, not a `Millisatoshi`, because it is a ratio, not an amount.

Reply:
Then what about multiplication? Keep in mind that a 'division with x' is the same as 'multiplication with 1/x' ... So to be consistent, we need to always return a Decimal/Float something (not Millisatoshi) for __mul__ __truediv__ and __floordiv__

Reply:
Meaning  Millisatoshi(5) * 3 := 15    (not msat)

Reply:
Millisatoshi(15) / Millisatoshi(5) := 3 (not msat)
So I multiple both side of the equation with Millisatoshi(5), and I get:
Millisatoshi(15) := 3 * Millisatoshi(5)
In your example:
Millisatoshi(5) * 3 := Millisatoshi(15)

Reply:
Sorry, I'm not quite getting it. In your opinion what should be the result of
`Millisatoshi(21) * Millisatoshi(2)`  be? Should it be just `42` or `Millisatoshi(42)? Currently it fails with an exception...

Reply:
The multiplication between Millisatoshis is difficult. In theory, it should be Millisatoshi square, Millisatoshi**2, Millisatoshi^2 or something like that.
I'm not sure, we need multiplication, maybe the exception is good. But I don't know.

Reply:
Also, can we write some more test? This should not raise exceptions:
- x = min(10, Millisatoshi(10))
- x = min(5 / 2, Millisatoshi(10))
- x = Millisatoshi(5 / 2)
- x = Millisatoshi(5) / 2

Reply:
> This should not raise exceptions:
> * x = Millisatoshi(5 / 2)

Not too sure about this. We ask it to initialize from `2.5` which, if no exception is raised, would likely remove the half msat. 

Maybe this whole class needs a rewrite/different approach. @cdecker already mentioned a couple of days ago that we might want to have a Decimal internally `self.millisatoshis` and only round down/floor when we do `self.to_json()`. This will keep precision when we do multiple div/mul operations after another... In this case we could go for the 2.5 in the constuctor as well :D

Reply:
`Millisatoshi(5) / 2` currently works. and the first two examples: `min(10, Millisatoshi(10))` and `min(5 / 2, Millisatoshi(10))`) should work, yes

Reply:
`Millisatoshi(5) / 2` currently works, and removes the half msat. So, I think `Millisatoshi(5 / 2)` should be the same.

Reply:
Now that you put it this way, Agreed.